Add release note for kubestash v2024.9.30 (#472)
appscode/blog: content/post/kubestash-v2024.9.30/index.md

        successfulJobsHistoryLimit: 1
        failedJobsHistoryLimit: 1
        jobTemplate:
          backoffLimit: 1
      repositories:
        - name: gcs-mysql-repo
          backend: gcs-backend
          directory: /mysql
          encryptionSecret:
           name: encrypt-secret
           namespace: demo
      addon:
        name: mysql-addon
        tasks:
          - name: logical-backup
            params:
              args: --single-transaction --set-gtid-purged=OFF
              databases: playground
```

***RestoreSession***

Here is an example of a `RestoreSession` object for restoring an externally managed `MySQL` database.

```yaml
apiVersion: core.kubestash.com/v1alpha1
kind: RestoreSession
metadata:
  name: mysql-restoresession
  namespace: demo
spec:
  target:
    apiGroup: appcatalog.appscode.com
    kind: AppBinding
    name: db-mysql-appbinding
    namespace: demo
  dataSource:
    repository: gcs-mysql-repo
    snapshot: latest
    encryptionSecret:
      name: encrypt-secret
      namespace: demo
  addon:
    name: mysql-addon
    tasks:
      - name: logical-backup-restore
```

### Improvements & Bug Fixes

#### Updated Declarative API 

> We’ve updated the declarative API for `BackupConfiguration` and `RestoreSession` in KubeStash. The `spec.sessions[*].timeout` field has been removed from `BackupConfiguration` and replaced with `spec.sessions[*].backupTimeout`, which sets the maximum duration for a backup. If the backup doesn’t finish within this time, it will be marked as failed. By default, no backup timeout is set.
> 
> Similarly, the `spec.timeout` field has been removed from `RestoreSession` and replaced with `spec.restoreTimeout`. This defines how long KubeStash should wait for a restore to complete before marking it as failed.

Reviewer: ```suggestion
We’ve updated the declarative API for `BackupConfiguration` and `RestoreSession` in KubeStash. The `spec.sessions[*].timeout` field has been removed from `BackupConfiguration` and replaced with `spec.sessions[*].backupTimeout`, which sets the maximum duration for the backup. KubeStash wraps the backup command(s) with this timeout. If the backup doesn’t finish within this time, the Snapshot and BackupSession should be marked as `Failed`. By default, no backup timeout is set.

Similarly, the `spec.timeout` field has been removed from `RestoreSession` and replaced with `spec.restoreTimeout` which sets the maximum duration for the restore. KubeStash wraps the restore command(s) with this timeout.  If the restore doesn’t finish within this time, the RestoreSession should be marked as `Failed`. By default, no restore timeout is set.
```

> #### Handle Backup Or Restore Timeout
> We’ve fixed a bug where the backup/restore job could remain active even after the deadline was exceeded (based on the timeout) with the `BackupSession` marked as failed. Now, KubeStash will wrap timeout with backup commands so the backup/restore job(s) will update `Snapshot` status with deadline exceeded error message and mark it as failed.

Reviewer: ```suggestion
```

> #### Handle Backup Or Restore Timeout
> We’ve fixed a bug where the backup/restore job could remain active even after the deadline was exceeded (based on the timeout) with the `BackupSession` marked as failed. Now, KubeStash will wrap timeout with backup commands so the backup/restore job(s) will update `Snapshot` status with deadline exceeded error message and mark it as failed.
> 

Reviewer: ```suggestion
KubeStash will now wrap the backup and restore commands with these timeouts, ensuring that Snapshot and RestoreSession statuses are updated with a 'deadline exceeded' error message.
```

> #### Handle Unexpected failures for Backup Or Restore Containers

Reviewer: ```suggestion
#### Handling Unexpected failures for Backup Or Restore

We’ve fixed a bug that caused the `BackupSession` to remain in the running phase when the backup/restore container failed unexpectedly with an error (i.e. OOMKill).

> ### Handle Content-Type YAML while Uploading to Object Storage

Reviewer: ```suggestion
#### Content-Type Handling for YAML Uploads to Object Storage
```
